test/lint/lint-circular-dependencies.sh

```diff
@@ -14,9 +14,12 @@ EXPECTED_CIRCULAR_DEPENDENCIES=(
     "blockchain/blockchain_genesis -> blockchain/blockchain_parameters -> blockchain/blockchain_genesis"
     "blockchain/blockchain_parameters -> settings -> key_io -> chainparams -> blockchain/blockchain_parameters"
     "blockdb -> validation -> injector -> blockdb"
+    "blockencodings -> injector -> p2p/graphene_receiver -> net_processing -> blockencodings"
```

Comment:
Did we introduced new circular deps? 😞

Reply:
Unfortunately. `injector` seems to be the bottleneck here: it's got a lot of dependencies, so if we use `GetComponent` in our code, we transitively depend on all that.

Reply:
This is just one half of the problem. There are two directions from every point on a circle.

The other direction is that a component uses non-component functions. This is why in proposer/staking code I abstracted all bitcoin capabilities behind `ActiveChain`, `Network`, etc. and also here the required capabilities from `net_processing` should probalby be added to `Network`.

Allows to mock dependencies, reduces circular dependencies, improves compile times.